src/main/java/bio/terra/service/upgrade/UpgradeService.java

```diff
@@ -31,6 +32,19 @@ private enum CustomFlight {
     TRANSACTION_SUPPORT(TransactionUpgradeFlight.class),
     CONVERT_DATASET_FILE_IDS(
         ConvertToPredictableFileIdsFlight.class,
+        request -> {
+          Preconditions.checkArgument(
+              request.getCustomArgs().size() == 1,
+              "Custom argument must have a single row: a valid dataset id");
+
+          Optional<UUID> datasetId = ValidationUtils.convertToUuid(request.getCustomArgs().get(0));
+          Preconditions.checkArgument(
+              datasetId.isPresent(), "Custom argument's single value is not a valid UUID");
+
+          return Map.of(JobMapKeys.DATASET_ID.getKeyName(), datasetId.get());
```

Comment:
I'm not familiar with `Preconditions` but if you had a function like it but handled `Optional<T>` you could instead write something like:
```
UUID datasetId = Preconditions.optionalPresent(ValidationUtils.convertToUuid(request.getCustomArgs().get(0),
    "Custom argument's single value is not a valid UUID");
return Map.of(JobMapKeys.DATASET_ID.getKeyName(), datasetId);
```
The goal being to avoid splitting up an `isPresent()` check from its corresponding `get()`.

Reply:
Oh need, I didn't realize preconditions had an optional handler built-in.

Reply:
It's not built-in, we'd have to write it. But it would be easy to do and improve the code IMO

Reply:
Something like
```
  static <T> T optionalPresent(Optional<T> optional, String message) {
    return optional.orElseThrow(() -> new IllegalArgumentException(message));
  }
```

Reply:
This chunk of code is following the existing pattern so I'd like to keep it for this time-sensitive PR. But, I'll create a follow-on to update the logic! 
https://broadworkbench.atlassian.net/browse/DR-3184